src/app/clusters/time-format-localization-server/time-format-localization-server.cpp

```diff
@@ -0,0 +1,203 @@
+using namespace chip::app::Clusters::TimeFormatLocalization::Attributes;
+using namespace chip::DeviceLayer;
+
+namespace {
+
+class TimeFormatLocalizationAttrAccess : public AttributeAccessInterface
+{
+public:
+    // Register for the Time Format Localization cluster on all endpoints.
+    TimeFormatLocalizationAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), TimeFormatLocalization::Id) {}
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+private:
+    CHIP_ERROR ReadSupportedCalendarTypes(AttributeValueEncoder & aEncoder);
+};
+
+TimeFormatLocalizationAttrAccess gAttrAccess;
+
+CHIP_ERROR TimeFormatLocalizationAttrAccess::ReadSupportedCalendarTypes(AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    DeviceLayer::AttributeList<CalendarType, DeviceLayer::kMaxCalendarTypes> supportedCalendarTypes;
+
+    if (DeviceLayer::PlatformMgr().GetSupportedCalendarTypes(supportedCalendarTypes) == CHIP_NO_ERROR)
+    {
+        err = aEncoder.EncodeList([&supportedCalendarTypes](const auto & encoder) -> CHIP_ERROR {
+            for (auto type : supportedCalendarTypes)
+            {
+                ReturnErrorOnFailure(encoder.Encode(type));
+            }
+
+            return CHIP_NO_ERROR;
+        });
+    }
+    else
+    {
+        err = aEncoder.EncodeEmptyList();
+    }
+
+    return err;
+}
+
+CHIP_ERROR TimeFormatLocalizationAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == TimeFormatLocalization::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    case SupportedCalendarTypes::Id:
+        return ReadSupportedCalendarTypes(aEncoder);
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+
+// Returns whether newType is a valid calendar type.  If it's not, validType is set to a valid calendar type,
+// if there are any, and to 0 if there are not.
+bool IsSupportedCalendarType(CalendarType newType, CalendarType & validType)
+{
+    DeviceLayer::AttributeList<CalendarType, DeviceLayer::kMaxCalendarTypes> supportedCalendarTypes;
+
+    if (DeviceLayer::PlatformMgr().GetSupportedCalendarTypes(supportedCalendarTypes) == CHIP_NO_ERROR)
+    {
+        for (auto type : supportedCalendarTypes)
+        {
+            validType = type;
+
+            if (validType == newType)
+            {
+                return true;
+            }
+        }
+    }
+    else
+    {
+        // Reset valid type if no supported calendar types found.
+        validType = CalendarType::kBuddhist;
+    }
+
+    return false;
+}
+
+} // anonymous namespace
+
+// =============================================================================
+// Pre-change callbacks for cluster attributes
+// =============================================================================
+
+static chip::Protocols::InteractionModel::Status
```

Comment:
A bunch of unaddressed comments from last time about unaddressed namespace prefixes.